kasa/discover.py

```diff
@@ -257,11 +270,11 @@ async def wait_for_discovery_to_complete(self):
         # Wait for any pending callbacks to complete
         await asyncio.gather(*self.callback_tasks)
 
-    def connection_made(self, transport) -> None:
+    def connection_made(self, transport: DatagramTransport) -> None:  # type: ignore[override]
```

Comment:
Without ignore, this complains that the super type class requires `BaseTransport`. The type is ignored similarly in https://github.com/python/typeshed/blob/main/stdlib/asyncio/protocols.pyi#L23 so I think this is fine.